React/Modules/RCTDevMenu.m

```diff
@@ -228,224 +178,45 @@ - (instancetype)init
     [commands registerKeyCommandWithInput:@"i"
                             modifierFlags:UIKeyModifierCommand
                                    action:^(__unused UIKeyCommand *command) {
-                                     [weakSelf.bridge.eventDispatcher
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated-declarations"
-                                      sendDeviceEventWithName:@"toggleElementInspector"
-                                      body:nil];
-#pragma clang diagnostic pop
+                                     [weakSelf.bridge.devSettings toggleElementInspector];
                                    }];
 
     // Reload in normal mode
     [commands registerKeyCommandWithInput:@"n"
                             modifierFlags:UIKeyModifierCommand
                                    action:^(__unused UIKeyCommand *command) {
-                                     weakSelf.executorClass = Nil;
+                                     [weakSelf.bridge.devSettings updateSettingWithValue:@(NO) forKey:kRCTDevSettingIsDebuggingRemotely];
```

Comment:
I prefer keeping it weaker (but with predefined key constants) for two reasons:
- Classes can define their own keys, such as `kRCTDevSettingShakeToShowDevMenu` which applies only to the RCTDevMenu UI but can still leverage the generic DevSettings interface.
- The documentation can live by the keys, instead of trying to put it in two places (the key const definition for, e.g. `kRCTDevSettingIsDebuggingRemotely` in addition to the method definition such as `setIsDebuggingRemotely:`)

However, if you disagree, then I can move the RCT_EXPORT'd methods from RCTDevSettings.m into RCTDevSettings.h, providing strongly-defined setters for these properties as you requested.